### ISSUE TITLE
Unable to download UTF-8-MAC file at aws-android-sdk-s3

### ISSUE THREAD

**Issue by TakkuMattsu** [aws-amplify/aws-sdk-android#360]
# Summary

* What service are you using?
  - [aws-android-sdk-s3](https://github.com/aws/aws-sdk-android/tree/master/aws-android-sdk-s3)
  - [aws-android-sdk-cognito](https://github.com/aws/aws-sdk-android/tree/master/aws-android-sdk-cognito)

* In what version of SDK are you facing the problem?
  - `com.amazonaws:aws-android-sdk-cognito:2.4.+`
  - `com.amazonaws:aws-android-sdk-s3:2.4.+`

* Is the issue limited to Simulators / Actual Devices?
  - Both

* Can your problem be resolved if you bump to a higher version of SDK?
   - No.
   - **But, It does not happen in `com.amazonaws:aws-android-sdk-s3:2.3.+`**
   - It seems to have occurred when fixing [this bug](https://github.com/aws/aws-sdk-android/blob/master/CHANGELOG.md#bug-fixes-12).(Amazon S3: fixed an issue which caused resource path to be incorrectly encoded.See AWS Forum Post)

# Test File

[だくてんありファイル.jpeg.zip](https://github.com/aws/aws-sdk-android/files/1425748/jpeg.zip)

![image](https://user-images.githubusercontent.com/3356758/32151660-fafaa912-bd61-11e7-8da8-5e3a2c0cbcd5.png)

```
Access Denied (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied
```

**Comment by minbi** [comment]
Thank you for reporting this issue. We will investigate and report back.

**Comment by kvasukib** [comment]
Sorry for the delayed reply. This issue has been fixed in 2.7.4. Please upgrade and check and let us know if you have any issues.

**Comment by kvasukib** [comment]
This is a duplicate of issue #526. We are tracking this issue in #526. We will capture all the details there.
